pallets/runtime/tests/src/identity_test.rs

```diff
@@ -1803,3 +1803,66 @@ fn add_investor_uniqueness_claim_v2_data(
         ),
     ]
 }
+
+fn setup_join_identity(source: &User, target: &User) {
+    assert_ok!(Identity::add_authorization(
+        source.origin(),
+        target.did.into(),
+        AuthorizationData::JoinIdentity(Permissions::default()),
+        None
+    ));
+    let auth_id = get_last_auth_id(&target.did.into());
+    assert_ok!(Identity::join_identity(target.did.into(), auth_id));
+}
+
+#[test]
+fn ext_join_identity_as_identity() {
+    ExtBuilder::default().build().execute_with(|| {
+        let alice = User::new(AccountKeyring::Alice);
+        let bob = User::new(AccountKeyring::Bob);
+
+        assert_noop!(
+            Identity::join_identity_as_identity(bob.origin(), 0),
+            "Authorization does not exist"
+        );
+
+        assert_ok!(Identity::add_authorization(
+            alice.origin(),
+            bob.did.into(),
+            AuthorizationData::Custom(Ticker::default()),
+            None
+        ));
+
+        let auth_id = get_last_auth_id(&bob.did.into());
+        assert_noop!(
+            Identity::join_identity_as_identity(bob.origin(), auth_id),
+            AuthorizationError::Invalid
+        );
+
+        setup_join_identity(&alice, &bob);
+    });
+}
+#[test]
+fn ext_leave_identity_as_identity() {
+    ExtBuilder::default().build().execute_with(|| {
+        let alice = User::new(AccountKeyring::Alice);
+        let bob = User::new(AccountKeyring::Bob);
+        let charlie = User::new(AccountKeyring::Charlie);
+
+        setup_join_identity(&alice, &bob);
+
+        let leave = |origin, did| {
+            assert_noop!(
+                Identity::leave_identity_as_identity(origin, did),
+                IdentityError::NotASigner
+            );
+        };
+
+        leave(alice.origin(), alice.did);
+        leave(charlie.origin(), alice.did);
+        assert_ok!(Identity::leave_identity_as_identity(
+            bob.origin(),
+            alice.did
+        ));
```

Comment:
What I had in mind was:
```suggestion
        let leave = |u: User| Identity::leave_identity_as_identity(u.origin(), alice.did);
        assert_noop!(leave(alice), IdentityError::NotASigner);
        assert_noop!(leave(charlie), IdentityError::NotASigner);
        assert_ok!(leave(bob));
```